### PR TITLE
Add set_secret() and get_secret() for iOS impl.

### LINKED ISSUE #187: IosCredentials is missing trait methods in v3
```
Showing All Messages
SEVERE:   --> /Users/muji/.cargo/registry/src/index.crates.io-6f17d22bba15001f/keyring-3.0.1/src/ios.rs:38:1
SEVERE:    |
SEVERE: 38 | impl CredentialApi for IosCredential {
SEVERE:    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `set_secret`, `get_secret` in implementation
SEVERE:    |
SEVERE:   ::: /Users/muji/.cargo/registry/src/index.crates.io-6f17d22bba15001f/keyring-3.0.1/src/credential.rs:24:5
SEVERE:    |
SEVERE: 24 |     fn set_secret(&self, password: &[u8]) -> Result<()>;
SEVERE:    |     ---------------------------------------------------- `set_secret` from trait
SEVERE: ...
SEVERE: 34 |     fn get_secret(&self) -> Result<Vec<u8>>;
SEVERE:    |     ---------------------------------------- `get_secret` from trait
SEVERE: 
SEVERE: For more information about this error, try `rustc --explain E0046`.
```

### PR DESCRIPTION
**Pull request by tmpfs**
Closes #187.

### CODE REVIEW

**Review by brotskydotcom — CHANGES_REQUESTED** [review]
Your definitions of `set_secret` and `get_secret` are perfect.

But, like the macOS implementation, you should have `set_password` call `set_secret` like this:
```
    fn set_password(&self, password: &str) -> Result<()> {
        self.set_secret(password.as_bytes())
    }
```
and you should have `get_password` call `get_secret` like this:
```
    fn get_password(&self) -> Result<String> {
        let password_bytes = self.get_secret()?;
        decode_password(password_bytes.to_vec())
    }
```

**Comment by tmpfs** [comment]
Good call @brotskydotcom , thats done!

**Review by brotskydotcom — APPROVED** [review]
Excellent! Thank you so much!

**Comment by tmpfs** [comment]
I notice you convert to `Vec<u8>` when calling `decode_password()` but `std::str::from_utf8` will take a slice so you can just pass to without the calls to `to_vec()`.

**Comment by brotskydotcom** [comment]
Can you please update README.md to add your name in the contributor's list?

**Comment by brotskydotcom** [comment]
> I notice you convert to `Vec<u8>` when calling `decode_password()` but `std::str::from_utf8` will take a slice so you can just pass to without the calls to `to_vec()`.
> 
> Edit: so if you prefer `std::str::from_utf8` to `String::from_utf8` then we could avoid the allocations here.

Well the definition of decode_password is shared across platforms and most of the others have already done the vector conversion.  But I still like the optimization you suggest in the iOS case, because password_bytes is already a Vec<u8> when it comes back from `get_secret`, so there's really no reason to have the `.to_vec()` redundant copy made.  It should just be `decode_password(password_bytes)`.

File: src/ios.rs [hunk 1]
```diff
@@ -52,11 +61,18 @@ impl CredentialApi for IosCredential {
     /// Returns a [NoEntry](ErrorCode::NoEntry) error if there is no
     /// credential in the store.
     fn get_password(&self) -> Result<String> {
-        let password_bytes =
-            get_generic_password(&self.service, &self.account).map_err(decode_error)?;
+        let password_bytes = self.get_secret()?;
         decode_password(password_bytes.to_vec())
```
Comment (brotskydotcom) [inline src/ios.rs:65]: Please remove this call, as it's unnecessary.  `password_bytes` is already a Vec<u8>.

**Review by brotskydotcom — CHANGES_REQUESTED** [review]
Just one last fix needed.

**Comment by brotskydotcom** [comment]
Whoops.  Just noticed one more fix needed to get the tests working on iOS:

You need to change line 91 from 
```
    use keyring::default::IosCredential;
```
to
```
    use keyring::ios::IosCredential;
```

File: src/ios.rs (hunk 1 above)
Comment (tmpfs) [reply src/ios.rs]: Agreed, thats done.

**Review by brotskydotcom — APPROVED** [review]
Perfect, thanks for keeping at it!

**Pull request merged**